
This patch explicitly links cx_Oracle against a system-supplied libodpic.so,
and obviates the requirement in the distributed code to download a copy of
ODPI-C for direct compilation into the module.

This patch has been suggested upstream, via 
https://github.com/oracle/python-cx_Oracle/issues/103


--- cx_Oracle-6.1/setup.py.orig	2017-12-13 16:21:25.305529042 -0800
+++ cx_Oracle-6.1/setup.py	2017-12-13 16:21:31.934733510 -0800
@@ -31,11 +31,41 @@
     extraCompileArgs.append("-DAIX5")
 elif sys.platform == "cygwin":
     extraLinkArgs.append("-Wl,--enable-runtime-pseudo-reloc")
 elif sys.platform == "darwin":
     extraLinkArgs.append("-shared-libgcc")
+elif sys.platform == "sunos5":
+    extraLinkArgs.append("-lodpic")
 
+
+depsrcs = ["src/BooleanVar.c", "src/Buffer.c", "src/Connection.c",
+           "src/Cursor.c", "src/CursorVar.c", "src/DateTimeVar.c",
+           "src/DeqOptions.c", "src/EnqOptions.c", "src/Error.c",
+           "src/IntervalVar.c", "src/LOB.c", "src/LobVar.c",
+           "src/LongVar.c", "src/MsgProps.c", "src/NumberVar.c",
+           "src/Object.c", "src/ObjectType.c", "src/ObjectVar.c",
+           "src/SessionPool.c", "src/StringVar.c", "src/Subscription.c",
+           "src/Variable.c"]
+
+if sys.platform != "sunos5":
+    # don't assume that we've got a system-delivered libodpic.so
+    depsrcs = depsrcs + ["odpi/include/dpi.h", "odpi/src/dpiImpl.h",
+                         "odpi/src/dpiConn.c", "odpi/src/dpiContext.c",
+                         "odpi/src/dpiData.c", "odpi/src/dpiDebug.c",
+                         "odpi/src/dpiDeqOptions.c", "odpi/src/dpiEnqOptions.c",
+                         "odpi/src/dpiEnv.c", "odpi/src/dpiError.c",
+                         "odpi/src/dpiGen.c", "odpi/src/dpiGlobal.c",
+                         "odpi/src/dpiHandlePool.c", "odpi/src/dpiLob.c",
+                         "odpi/src/dpiMsgProps.c", "odpi/src/dpiObject.c",
+                         "odpi/src/dpiObjectAttr.c", "odpi/src/dpiObjectType.c",
+                         "odpi/src/dpiOci.c", "odpi/src/dpiOracleType.c",
+                         "odpi/src/dpiPool.c", "odpi/src/dpiRowid.c",
+                         "odpi/src/dpiStmt.c", "odpi/src/dpiSubscr.c",
+                         "odpi/src/dpiUtils.c", "odpi/src/dpiVar.c"]
+    extraCompileArgs.append("-DODPI_NOT_PRESENT=1")
+
+
 class test(distutils.core.Command):
     description = "run the test suite for the extension"
     user_options = []
 
     def finalize_options(self):
@@ -71,30 +101,11 @@
         name = "cx_Oracle",
         include_dirs = ["odpi/include", "odpi/src"],
         extra_compile_args = extraCompileArgs,
         extra_link_args = extraLinkArgs,
         sources = ["src/cx_Oracle.c"],
-        depends = ["src/BooleanVar.c", "src/Buffer.c", "src/Connection.c",
-                "src/Cursor.c", "src/CursorVar.c", "src/DateTimeVar.c",
-                "src/DeqOptions.c", "src/EnqOptions.c", "src/Error.c",
-                "src/IntervalVar.c", "src/LOB.c", "src/LobVar.c",
-                "src/LongVar.c", "src/MsgProps.c", "src/NumberVar.c",
-                "src/Object.c", "src/ObjectType.c", "src/ObjectVar.c",
-                "src/SessionPool.c", "src/StringVar.c", "src/Subscription.c",
-                "src/Variable.c", "odpi/include/dpi.h", "odpi/src/dpiImpl.h",
-                "odpi/src/dpiConn.c", "odpi/src/dpiContext.c",
-                "odpi/src/dpiData.c", "odpi/src/dpiDebug.c",
-                "odpi/src/dpiDeqOptions.c", "odpi/src/dpiEnqOptions.c",
-                "odpi/src/dpiEnv.c", "odpi/src/dpiError.c",
-                "odpi/src/dpiGen.c", "odpi/src/dpiGlobal.c",
-                "odpi/src/dpiHandlePool.c", "odpi/src/dpiLob.c",
-                "odpi/src/dpiMsgProps.c", "odpi/src/dpiObject.c",
-                "odpi/src/dpiObjectAttr.c", "odpi/src/dpiObjectType.c",
-                "odpi/src/dpiOci.c", "odpi/src/dpiOracleType.c",
-                "odpi/src/dpiPool.c", "odpi/src/dpiRowid.c",
-                "odpi/src/dpiStmt.c", "odpi/src/dpiSubscr.c",
-                "odpi/src/dpiUtils.c", "odpi/src/dpiVar.c"])
+        depends = depsrcs)
 
 # perform the setup
 setup(
         name = "cx_Oracle",
         version = BUILD_VERSION,
